server/etcdserver/server.go

```diff
@@ -764,6 +765,10 @@ func (s *EtcdServer) run() {
 	if err != nil {
 		lg.Panic("failed to get snapshot from Raft storage", zap.Error(err))
 	}
+	fi, err := s.r.raftStorage.FirstIndex()
```

Comment:
Please don't use short obfuscated variable name for variables spanning hundred lines of code. Either:
* Move the call to FirstIndex next to creation of etcdProgress
* Use proper variable name like `firstRaftIndex`.